### ISSUE TITLE
How to meson-ify LuaJIT 2.1.0 beta3 / master branch?

### ISSUE THREAD

**Issue by kvnxiao** [franko/luajit#1]
Hi @franko !

I'm trying to meson-ify the latest git version of LuaJIT from the git mirror here https://github.com/LuaJIT/LuaJIT which is master branch on 2.1.0 and am having troubles getting it working.

Would you mind pointing me in the right direction how I could achieve this basing it off your current repository changes?

**Comment by kvnxiao** [comment]
Seems like I got it updated and working (compiled on Windows 10 build 19041.573 using TDM-GCC 9.2.0).

The following changes were required:
1. Add in the new source files for `ljcore_sources` variable in `src/meson.build` (2.1.0 introduced 3-4 new `*.c` files)
2. Modify `lj_arch_test.c` to change dasm_arch to `x64` on LJ_TARGET_X64:
	```c
	#ifdef LJ_TARGET_X64
	    dasm_arch = "x64";
	#else
	    dasm_arch = lj_arch;
	#endif
	```
3. Add new endian-ness flag to `lj_arch_test.c`:
	```c
	#ifdef LJ_LE
	    add_def(dasm, &dasm_n, "ENDIAN_LE");
	#else
	    add_def(dasm, &dasm_n, "ENDIAN_BE");
	#endif
	```

And this compiled without issues:

![image](https://user-images.githubusercontent.com/14130413/96354705-962c4a80-10a7-11eb-9e7b-67b5c6d3262e.png)

**Comment by franko** [comment]
Hi @kvnxiao , looks cool, thank you for sharing this.

I tried your changes and it compiles but on Linux 64bit it crashes on startup. You may have a look, there is now a v2.1 branch in the repository that you can test. It just have your suggested changes for the Meson build.

**Comment by franko** [comment]
Ok, I fixed the 'v2.1' branch, I just wrongly added the LUAJIT_DISABLE_GC64 option. Now seems to be fine.

**Comment by kvnxiao** [comment]
Awesome! Sounds good, and I'm glad that the information was useful.

Going to close this issue now!
